docs/source/explanations/architecture.md

```diff
@@ -0,0 +1,90 @@
+Database** (PostgreSQL or SQLite) is used to store session information and to
+validate API keys.
+
+Endpoints that serve metadata or data resolve the URL path to identify the
+relevant **Adapter** (about which, more below), which returns the data
+as a scientific data structure. It may be a "lazy" data structure,
+representing data that will be loaded later, on demand, piecemeal.
+
+The endpoint implements [content negotiation][], comparing the list of formats
+and accepted by the client to those supported by the server for this particular
+dataset. It dispatches to a registry of **Serializers**, which convert the data
+structure into bytes which can be sent in a response by FastAPI. Custom
+serializers may be registered in server configuration.
+
+On its way, the response may be compressed, again using content negotiation
+to compare the list of compression schemes supported by the client (if any)
+to those the server deems appropriate to the dataset.
+
+## Adapter
+
+In Tiled, an **Adapter** provides a standard interface to data, regardless of
+how it is stored. Adapters that wrap different structure have different
+interfaces. For example, an array adapter implements `read_block` whereas a
+table adapter implements `read_partition`. But all array adapters are alike,
+and all table adapters are alike. They enable the server to abstract over
+the details of _how_ to get the data.
+
+Typically Adapters take a filepath or URI and, perhaps, some additional
+configuration. For development, test, or demo purposes, Adapters can wrap data
+in memory, like a numpy array. Several Adapters are included in the Tiled
+codebase, in the spirit of "batteries included," but Adapters can be defined in
+external modules, too, and operate on the same footing as the built-in ones.
+
+## Catalog
+
+The Catalog is an Adapter that stores the metadata and structure for a
+potentially large number of datasets in a SQL database (PostgreSQL or SQLite).
+This enables it to efficiently response to _metadata_ requests and _search_
+requests without opening any data files. Requests for _data_ are then
+dispatched down to the appropriate Adapter which can load the data
+from the given storage medium and format.
+
+Not all Tiled servers are configured to use the Catalog:
+
+* Demo deployments wrapping a handful of datasets use data Adapters directly,
+  with no need for a database.
+* Specialized deployments with custom Adapters may use a custom database or
+  external service.
+
+But for most standard applications, including serving a directory of files or
+providing a writable data store, the Catalog is used.
+
+[FastAPI]: https://fastapi.tiangolo.com/
+[httpx]: https://www.python-httpx.org/
+[Starlette]: https://www.starlette.io/
+[OpenAPI]: https://www.openapis.org/
+[Pydantic]: https://docs.pydantic.dev/
+[content negotiation]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation
```

Comment:
This is a nice paradigm for tidying links in the docs.  I haven't seen it before.